controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java

```diff
@@ -130,12 +131,15 @@ public static VersionStatus compute(Controller controller) {
         ListMap<Version, HostName> versions = new ListMap<>();
         for (ZoneId zone : zones) {
             for (SystemApplication application : SystemApplication.all()) {
-                if (!application.configConvergedIn(zone, controller)) {
-                    throw new IllegalStateException("Config for " + application.id() + " in " + zone + " has not converged");
+                boolean configConverged = application.configConvergedIn(zone, controller);
+                if (!configConverged) {
+                    log.log(LogLevel.WARNING, "Config for " + application.id() + " in " + zone + " has not converged");
                 }
                 for (Node node : controller.configServer().nodeRepository().list(zone, application.id(),
                                                                                  SystemApplication.activeStates())) {
-                    versions.put(node.currentVersion(), node.hostname());
+                    // Only use current node version if config has converged
+                    Version nodeVersion = configConverged ? node.currentVersion() : controller.systemVersion();
+                    versions.put(nodeVersion, node.hostname());
```

Comment:
A pity we can't do this on a per-node basis.

Reply:
This may give strange behaviour, where a node appears to downgrade. Could this be a problem?

Reply:
Discussed offline.